binds/python/bind_immutable.cpp

```diff
@@ -20,10 +20,16 @@ static void bind_immutable_module(py::module &m) {
     py::add_ostream_redirect(m, "ostream_redirect");
 
     py::class_<morphio::Morphology>(m, "Morphology")
-        .def(py::init<const morphio::URI&, unsigned int>(),
+        .def(py::init<const std::string&, unsigned int>(),
              "filename"_a, "options"_a=morphio::enums::Option::NO_MODIFIER)
         .def(py::init<morphio::mut::Morphology&>())
-
+        .def(py::init([](py::object arg, unsigned int options) {
+                          return std::unique_ptr<morphio::Morphology>(
+                              new morphio::Morphology(py::str(arg), options)
```

Comment:
what happens when `py::str()` doesn't work?

Reply:
I don't manage to find an example of object for which it does not work.
```
class A:
    pass
morphio.Morphology(A())
```
The error is `RawDataError: File: <__main__.A object at 0x7fc49a349470> does not exist.`
Because `str(A())` still returns something even though `str` is not explicitly defined.
But I think the error message is still helpful.

Reply:
yeah, on further thought, `str()` is basically defined for everything